connectivity/builder/no_interrupted_connections.go

```diff
@@ -13,5 +15,15 @@ type noInterruptedConnections struct{}
 func (t noInterruptedConnections) build(ct *check.ConnectivityTest, _ map[string]string) {
 	newTest("no-interrupted-connections", ct).
 		WithCondition(func() bool { return ct.Params().IncludeConnDisruptTest }).
-		WithScenarios(tests.NoInterruptedConnections())
+		WithScenarios(tests.NoInterruptedConnections()).
+		WithFinalizer(func(ctx context.Context) error {
+			// Delete the test-conn-disrupt pods immediately after the test run
+			// has finished to reduce CPU consumption (the pods generate a lot
+			// of traffic, and in most CI jobs Cilium runs with monitor
+			// aggregation disabled).
+			if !ct.Params().ConnDisruptTestSetup {
+				return ct.DeleteConnDisruptTestDeployment(ctx, ct.K8sClient())
```

Comment:
Should iterate over all clients, to ensure that the clean-up happens correctly also in a clustermesh environment.

Reply:
Oops, I keep forgetting about these :+1: